test/test_e2e.rb

```diff
@@ -104,4 +112,20 @@ def test_spec_file_splitting
       "./spec/fast_spec.rb",
     ], queue)
   end
+
+  def test_suite_with_failures_and_fail_fast
+    queue = exec_build("failing_suite", "--fail-fast 1")
+
+    assert_equal 1, queue.fail_fast
+    assert queue.build_failed_fast?
+    refute queue.build_successful?
+    assert_equal queue.fail_fast, queue.example_failures.length +
+                                  queue.non_example_errors.length
+
+    # 1 <= unprocessed_jobs <= 2
+    # Either Success, Fail (after N requeues), or Fail (after N requeues)
+    assert_includes [1, 2], queue.unprocessed_jobs.length
+
+    assert_includes [1, 2], queue.unprocessed_jobs.length
```

Comment:
This assertion is wrong I believe. We should assert on `processed_jobs`, right?

Reply:
woops! It must have slipped from a copy paste

Reply:
I will wait for CI to finish the build and we 're good to go